Entities/AtomGroup.h

```diff
@@ -1,662 +1,387 @@
+
+		/// <summary>
+		/// Makes this AtomGroup travel as a lifeless limb, constrained to a radius around the joint pin in the center.
+		/// </summary>
+		/// <param name="ownerPos">A Vector with the world position of the owner MOSRotating.</param>
+		/// <param name="jointOffset">A Vector with the rotated offset of the joint that this should flail around in a radius.</param>
+		/// <param name="limbRadius">The radius/range of the limb this is to simulate.</param>
+		/// <param name="velocity">A Vector with the velocity of the owning MOSRotating.</param>
+		/// <param name="angVel">A float with the angular velocity in rad/s of the owning MOSRotating.</param>
+		/// <param name="mass">The mass of this dead weight limb.</param>
+		/// <param name="travelTime">The amount of time in seconds that this AtomGroup is supposed to travel.</param>
+		void FlailAsLimb(const Vector &ownerPos, const Vector &jointOffset, const float limbRadius, const Vector &velocity, const float angVel, const float mass, const float travelTime);
+#pragma endregion
+
+#pragma region Collision
+		/// <summary>
+		/// Adds a MOID that this AtomGroup should ignore collisions with during its next Travel sequence.
+		/// </summary>
+		/// <param name="moidToIgnore">The MOID to add to the ignore list.</param>
+		void AddMOIDToIgnore(MOID moidToIgnore) { m_IgnoreMOIDs.push_back(moidToIgnore); }
+
+		/// <summary>
+		/// Checks whether this AtomGroup is set to ignore collisions with a MO of a specific MOID.
+		/// </summary>
+		/// <param name="whichMOID">The MOID to check if it is ignored.</param>
+		/// <returns>Whether or not this MOID is being ignored.</returns>
+		bool IsIgnoringMOID(MOID whichMOID) { return (*(m_Atoms.begin()))->IsIgnoringMOID(whichMOID); }
+
+		/// <summary>
+		/// Clears the list of MOIDs that this AtomGroup is set to ignore collisions with during its next Travel sequence.
+		/// This should be done each frame so that fresh MOIDs can be re-added. (MOIDs are only valid during a frame).
+		/// </summary>
+		void ClearMOIDIgnoreList() { m_IgnoreMOIDs.clear(); }
+
+		/// <summary>
+		/// Gets whether any of the Atoms in this AtomGroup are on top of terrain pixels.
+		/// </summary>
+		/// <returns>Whether any Atom of this AtomGroup is on top of a terrain pixel.</returns>
+		bool InTerrain() const;
+
+		/// <summary>
+		/// Gets the ratio of how many Atoms of this AtomGroup that are on top of intact terrain pixels.
+		/// </summary>
+		/// <returns>The ratio of Atoms on top of terrain pixels, form 0 to 1.0.</returns>
+		float RatioInTerrain() const;
+
+		/// <summary>
+		/// Checks whether any of the Atoms in this AtomGroup are on top of terrain pixels, and if so, attempt to move the OwnerMO out so none of the Atoms are inside any terrain pixels anymore.
+		/// </summary>
+		/// <param name="position">Current position of the owner MO.</param>
+		/// <param name="strongerThan">Only attempt to move out of materials stronger than this specific ID.</param>
```

Comment:
I would rename this parameter so it's clearer what it means without needing the comment. `MinimumStrengthMaterial` or something might work. If we can't figure out a better name, so be it.

Reply:
I don't think that's better honestly, and I don't have anything better than that either.

Reply:
Fair, I don't have a better name off the top of my head either, I guess it's fine as-is.